core/src/main/java/io/questdb/griffin/engine/functions/rnd/RndStringMemory.java

```diff
@@ -28,27 +28,32 @@
 import io.questdb.cairo.vm.Vm;
 import io.questdb.cairo.vm.api.MemoryAR;
 import io.questdb.griffin.SqlException;
+import io.questdb.std.DeepCloneable;
 import io.questdb.std.MemoryTag;
 import io.questdb.std.Misc;
 import io.questdb.std.Rnd;
 
 import java.io.Closeable;
 
-class RndStringMemory implements Closeable {
+class RndStringMemory implements Closeable, DeepCloneable<RndStringMemory> {
```

Comment:
We don't need to clone these objects: that's because they're only used in `rnd_*` functions and those functions aren't deterministic, so shouldn't be used in parallel execution.